src/main/java/io/github/intoto/models/Subject.java

```diff
@@ -0,0 +1,72 @@
+package io.github.intoto.models;
+
+import java.util.Map;
+import java.util.Objects;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+
+/**
+ * Set of software artifacts that the attestation applies to. Each element represents a single *
+ * software artifact.
+ */
+public class Subject {
+  /**
+   * Identifier to distinguish this artifact from others within the subject.
+   *
+   * <p>The semantics are up to the producer and consumer. Because consumers evaluate the name
+   * against a policy, the name SHOULD be stable between attestations. If the name is not
+   * meaningful, use "_". For example, a SLSA Provenance attestation might use the name to specify
+   * output filename, expecting the consumer to only considers entries with a particular name.
+   * Alternatively, a vulnerability scan attestation might use the name "_" because the results
+   * apply regardless of what the artifact is named.
+   *
+   * <p>MUST be non-empty and unique within subject.
+   */
+  @NotBlank(message = "subject name must not be blank")
+  private String name;
+  /**
+   * Collection of cryptographic digests for the contents of this artifact.
+   *
+   * <p>Two DigestSets are considered matching if ANY of the fields match. The producer and consumer
+   * must agree on acceptable algorithms. If there are no overlapping algorithms, the subject is
+   * considered not matching.
+   */
+  @NotEmpty(message = "digest must not be empty")
+  private Map<
+          @NotBlank(message = "digest key contents can be empty strings") String,
+          @NotBlank(message = "digest value contents can be empty strings") String>
+      digest;
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public Map<String, String> getDigest() {
```

Comment:
optional: create a standalone type called DigestSet (https://github.com/in-toto/attestation/blob/main/spec/field_types.md#DigestSet) and move DigestSetAlgorithmType into an inner enum of that class?

Reply:
Making the key a DigestSetAlgorithmType would prevent new types of algorithms to be added in the future.

Reply:
That's right!  I'd asked you to make it this way in the first place.  WDYT about adding a comment that explains that?

Reply:
Done.